### PR TITLE
FIXING IDENTITY-3659 and CARBON-15457

### PR DESCRIPTION
**Pull request by dracusds123**

### CODE REVIEW

**Comment by callkalpa** [comment]
Hi @dracusds123,

We need a carbon JIRA and the fix needs to be reviewed.

**Comment by johannnallathamby** [comment]
Reviewed. Can merge.

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt.ui/src/main/resources/web/tenant-mgt/js/tenant_config.js [hunk 1]
```diff
@@ -27,8 +27,8 @@ function addTenant(isUpdating, isPublicCloud, isEmailUser) {
             if (adminPassword.value != adminPasswordRepeat.value) {
                 reason += jsi18n["password.mismatched"];
             }
-            if (adminPassword.value.length < 6) {
-                reason += jsi18n["password.length"];
+            if (validateString(adminPassword, passwordRegex) != "") {
+                reason += "Password policy violation. Please enter your password to follow the regex " + passwordRegex.value;
```
Comment (darshanasbg) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt.ui/src/main/resources/web/tenant-mgt/js/tenant_config.js]: Need to retrieve reason string from i18n.

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt.ui/src/main/resources/web/tenant-mgt/js/tenant_config.js [hunk 2]
```diff
@@ -90,8 +90,8 @@ function addTenant(isUpdating, isPublicCloud, isEmailUser) {
...
```
Comment (darshanasbg) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt.ui/src/main/resources/web/tenant-mgt/js/tenant_config.js]: Need to retrieve reason string from i18n.

**Comment by darshanasbg** [comment]
Hi @callkalpa,

Can you please merge this PR?

Thanks,

**Pull request merged**